tests/unittest/_torch/modeling/test_nano_encode_extractor.py

```
@@ -0,0 +1,641 @@
+        # A multi-modality param with NO multimodal_item_order key cannot build a
+        # transient MixedModalItemOrder (from_metadata returns None), so the
+        # per-item extractor raises a clear error naming both the typed view it
+        # could not build and the missing wire key it needs.
+        param = _make_param(
+            {
+                "modality_type": ["image", "audio"],
+                "image": {"pixel_values": "imgs"},
+                "audio": {"input_audio_features": "auds"},
+            }
+        )
+        with pytest.raises(KeyError, match="MixedModalItemOrder.*multimodal_item_order"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)